tests/add_tests.rs

```diff
@@ -75,6 +77,29 @@ async fn add_functionality() {
     ));
 }
 
+/// Test adding a package with a specific channel
+#[tokio::test]
+async fn add_with_channel(){
+    let pixi = PixiControl::new().unwrap();
+
+    pixi.init()
+        .no_fast_prefix_overwrite(true)
+        .await
+        .unwrap();
+
+    pixi.add("conda-forge::py_rattler").without_lockfile_update().await.unwrap();
```

Comment:
Currently still a slow test as there is a bug in this `--no-lockfile-update` so it is actually solving. This is fixed in: https://github.com/prefix-dev/pixi/pull/1683

Reply:
Otherwise I would have used the mock PackageDatabase :wink: